cli/cli/helpers/portal_manager/binary_artifact_getter.go

```diff
@@ -0,0 +1,166 @@
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Unable to get list of assets from latest version")
+	}
+	detectedArchitecture := fmt.Sprintf("%s_%s", runtime.GOOS, runtime.GOARCH)
+	assetFileExpectedSuffix := fmt.Sprintf("%s%s", detectedArchitecture, githubAssetExtension)
+	var releaseAssetToUse *github.ReleaseAsset
+	for _, releaseAsset := range allReleaseAssets {
+		if strings.HasSuffix(releaseAsset.GetName(), assetFileExpectedSuffix) {
+			releaseAssetToUse = releaseAsset
+		}
+	}
+	if releaseAssetToUse == nil {
+		return nil, stacktrace.NewError("Unable to find Kurtosis Portal binary matching the current architecture. Detected architecture was: '%s'", detectedArchitecture)
+	}
+	logrus.Debugf("Kurtosis Portal binary found: '%s' with ID '%d'", releaseAssetToUse.GetName(), releaseAssetToUse.GetID())
+
+	// Download the content of the asset. It is expected to be a .tar.gz file containing the Kurtosis Portal binary
+	artifactContent, _, err := ghClient.Repositories.DownloadReleaseAsset(ctx, kurtosisTechGithubOrg, kurtosisPortalGithubRepoName, releaseAssetToUse.GetID(), http.DefaultClient)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Unable to download Kurtosis portal binary. Name was: '%s', ID was: '%d'", releaseAssetToUse.GetName(), releaseAssetToUse.GetID())
+	}
+	return artifactContent, nil
+}
+
+func extractBinaryToFile(assetContent io.ReadCloser, assetVersion string, destFilePath string, destVersionFilePath string) error {
+	gzipReader, err := gzip.NewReader(assetContent)
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to open a GZIP reader on the asset content")
+	}
+	defer gzipReader.Close()
+
+	assetTarReader := tar.NewReader(gzipReader)
+	onlyFileHeader, err := assetTarReader.Next()
+	if err != nil || onlyFileHeader == nil {
+		return stacktrace.Propagate(err, "Asset archive seems to be empty, this is unexpected")
+	}
+	if onlyFileHeader.Typeflag != tar.TypeReg {
+		return stacktrace.Propagate(err, "Archive seems to be a directory, but expecting a single file")
+	}
+
+	portalBinaryFile, err := os.Create(destFilePath)
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to create a new executable file to store Kurtosis Portal binary")
+	}
+	if err = os.Chmod(portalBinaryFile.Name(), portalBinaryFileMode); err != nil {
+		return stacktrace.Propagate(err, "Unable to create a new executable file to store Kurtosis Portal binary")
+	}
+
+	if _, err = io.Copy(portalBinaryFile, assetTarReader); err != nil {
+		return stacktrace.Propagate(err, "Unable to copy content of Kurtosis Portal binary to executable file")
+	}
+
+	if err = os.WriteFile(destVersionFilePath, []byte(assetVersion), portalVersionFileMode); err != nil {
+		return stacktrace.Propagate(err, "Portal binary file successfully stored but an error occurred persisting its corresponding version.")
+	}
+	logrus.Debugf("Kurtosis Portal binary downloaded to '%s'", destFilePath)
```

Comment:
1. Works on Mac, at least. Will try on linux on the CI. But since in both case we use GO to generate the asset and here retrieve the variable, I'd hope internally it uses GOOS and GOARCH. Anyway, will try out of cautiousness. I am planning on adding integration tests for all this afterwards anyway, which will run on Linux in CI
2. Yeah, I was thinking about this, and same as you I wasn't able to land on what the best option is. So, I chose to default to upgrade automatically because it's less things users have to think about, and also I don't see any major problem with automating the upgrade. If we introduce incompatibilities, we can always add a version compatibiility table somewhere and improve the automatic download logic